src/random.cpp

```diff
@@ -479,6 +438,13 @@ class RNGState {
             hasher.Finalize(buf);
             // Store the last 32 bytes of the hash output as new RNG state.
             memcpy(m_state, buf + 32, 32);
+            // Handle requests for deterministic randomness.
+            if (allow_deterministic && m_deterministic_prng.has_value()) {
```

Comment:
Does `allow_deterministic=false` and `m_deterministic_prng.has_value()=true` imply that something has gone wrong? Should we log an error or take some action or are there cases where this is expected behavior?